torch/fx/passes/fake_tensor_prop.py

```diff
@@ -30,6 +30,7 @@ def __init__(self, module: torch.fx.GraphModule, mode: Optional[FakeTensorMode]
             mode = FakeTensorMode()
         self._mode = mode
         mode.epoch += 1
+        mode.reset_nt_tensor_id_counter()
```

Comment:
just thinking: would it be correct / less error prone just to reset this directly in `FakeTensorMode.__enter__`, so we don't have to worry about doing it manually?

Reply:
This is a good question. - I do remember trying this earlier and I logging when enter/exit would be called, it appears that we enter/exit every time we dispatch into any op in FakeTensorMode's torch dispatch. Would need to double check if there is a outer context manager that is still enabled while all of this is happening (maybe it is fine to rely on enter/exit if that is the case?)

Reply:
Okay, I added a breakpoint to `__enter__`'s  is a no-op path, and it is never reached. I'm not sure we can rely on enter/exit to do the right thing.

Reply:
yeah... i think the fake tensor caching logic re-enables/disables the fake mode so maybe this is a pain :/